### PR TITLE
Drop unused tables & unneeded access token ID for events.

### LINKED ISSUE #15786: Gracefully drop MSC2716 related tables (incrementally import history)
Gracefully drop MSC2716 related tables (incrementally import history) in multiple steps/phases.

### Step 1

In one Synapse version, stop writing to any MSC2716 related tables. The normal [steps to gracefully drop a table](https://github.com/matrix-org/synapse/blob/fcc3ca37e1b404981d9a0d6f2708e14407775b97/docs/development/database_schema.md) usually include a step to first switch over reads but keep writing but since this is a experimental feature, we don't need to worry about the feature working we can just stop reading and writing in the same step.

This is being tackled in https://github.com/matrix-org/synapse/pull/15748

### Step 2

In the Synapse release after the first step ships, we can bump `SCHEMA_COMPAT_VERSION` and drop the tables altogether.

Relevant tables to drop:

```sql
-- Drop tables related to MSC2716 since the implementation is being removed
DROP TABLE insertion_events;
DROP TABLE insertion_event_edges;
DROP TABLE insertion_event_extremities;
DROP TABLE batch_events;
```

### LINKED ISSUE #16042: Clean-up event_txn_id table
1. In a future version we need to stop reading & writing to the `event_txn_id` table & bump the schema version.
2. In a future future version, bump the minimum schema version to the new version in step 1 & drop the `event_txn_id` table.
3. In step 2, we can also stop writing the access token ID into the internal metadata for events.

See https://github.com/matrix-org/synapse/pull/15629#discussion_r1279686691

### PR DESCRIPTION
**Pull request by clokep**
Fixes #16042 and fixes #15786.

### CODE REVIEW

File: synapse/storage/databases/main/purge_events.py [hunk 1]
```diff
@@ -450,10 +450,6 @@ def _purge_room_txn(self, txn: LoggingTransaction, room_id: str) -> List[int]:
             "e2e_room_keys",
             "event_push_summary",
             "pusher_throttle",
-            "insertion_events",
-            "insertion_event_extremities",
-            "insertion_event_edges",
-            "batch_events",
```
Comment (clokep) [inline synapse/storage/databases/main/purge_events.py:456]: I guess we really need to do this in an earlier PR?

Comment (clokep) [reply synapse/storage/databases/main/purge_events.py:456]: See #16273.

**Comment by clokep** [comment]
#16273 will be in v1.93.0, so we'll want to wait a couple more versions until we can land this.

**Comment by clokep** [comment]
`develop` is now v1.95.0's base so I think we can merge this.

**Review by erikjohnston — APPROVED** [review]

**Pull request merged**